aio/tools/transforms/angular-base-package/processors/generateKeywords.js

```diff
@@ -120,63 +114,64 @@ module.exports = function generateKeywordsProcessor(log, readFilesProcessor) {
   };
 };
 
-
 function isString(value) {
   return typeof value == 'string';
 }
 
-function convertToMap(collection) {
-  const obj = {};
-  collection.forEach(key => { obj[key] = true; });
-  return obj;
-}
-
-// If the heading contains a name starting with ng, e.g. "ngController", then add the
-// name without the ng to the text, e.g. "controller".
-function tokenize(text) {
-  const rawTokens = text.split(/[\s\/]+/mg);
+function tokenize(text, ignoreWords, dictionary) {
+  // Split on whitespace and things that are likely to be HTML tags (this is not exhaustive but reduces the unwanted tokens that are indexed).
+  const rawTokens = text.split(/[\s\/]+|<\/?[a-z]+(?:\s+\w+(?:="[^"]+")?)*>/img);
   const tokens = [];
-  rawTokens.forEach(token => {
+  for(let token of rawTokens) {
+    token = token.trim();
+
     // Strip off unwanted trivial characters
-    token = token
-        .trim()
-        .replace(/^[_\-"'`({[<$*)}\]>.]+/, '')
-        .replace(/[_\-"'`({[<$*)}\]>.]+$/, '');
-    // Ignore tokens that contain weird characters
-    if (/^[\w.\-]+$/.test(token)) {
-      tokens.push(token.toLowerCase());
-      const ngTokenMatch = /^[nN]g([A-Z]\w*)/.exec(token);
-      if (ngTokenMatch) {
-        tokens.push(ngTokenMatch[1].toLowerCase());
-      }
+    token = token.replace(/^[_\-"'`({[<$*)}\]>.]+/, '').replace(/[_\-"'`({[<$*)}\]>.]+$/, '');
+
+    // Skip if in the ignored words list
+    if (ignoreWords.has(token.toLowerCase())) {
+      continue;
+    }
+
+    // Skip tokens that contain weird characters
+    if (!/^[\w._-]+$/.test(token)) {
```

Comment:
Nit: `\w` includes `_`, so that is redundant:

```suggestion
    if (!/^[\w.-]+$/.test(token)) {
```

Reply:
One downside of the final simplified version is that it allows the empty string through.